### PR TITLE
Replace custom (SWT)ResourceManager from eclipse.tips

### PR DESCRIPTION
**Pull request by HannesWell**
Instead of using a custom `(SWT)ResourceManager` (IIRC WindowBuilder generates it?), use JFace' standard `ResourceManager` to cache `SWT` resources.

@wimjongman, according to the git history, you created that part of the code. Do you want to review this?
@iloveeclipse since you told that you have a large help-system for your product, maybe you also have your own tips and are interested in this as well.

### CODE REVIEW

**Comment by HannesWell** [comment]
This would benefit from https://github.com/eclipse-platform/eclipse.platform.ui/pull/838.

**Review by wimjongman — APPROVED** [review]
Hey Hannes, Sure this looks good to me. Thanks for maintaining the Tips FW.

File: org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java [hunk 1]
```diff
@@ -43,12 +45,14 @@ public class TipDialog extends Dialog {
 	private TipComposite fTipComposite;
 	private int fShellStyle;
 	private IDialogSettings fDialogSettings;
+	private ResourceManager resourceManager;
```
Comment (jukzi) [inline org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java:48]: resourceManagers need to be disposed or we get non-disposed errors

**Review by jukzi — CHANGES_REQUESTED** [review]

**Comment by vogella** [comment]
Maybe WindowBuilder should retire this support for its own resource manager and switch to use the standard JFace one. cc @ptziegler

**Comment by ptziegler** [comment]
> Maybe WindowBuilder should retire this support for its own resource manager and switch to use the standard JFace one. 

Hm, at first glance, there seems to be a lot of overlap between the JFace resource manager and the WindowBuilder resource manager. 
The real issue is going to be the decorateImage() method, which we use to quickly construct composite images. At the very least, we should extend the JFace class in order to reduce the amount of duplicate code.

**Comment by HannesWell** [comment]
I have now reworked this a bit further to improve the caching of slider-images. Before all images ever created in a re-paint were kept in the cache due to missing/improper `hashcode/equals`. They were disposed in the end of each paint (so it was known), but nevertheless the cache was filled over time.
I have solved this with a custom ImageDescriptor subclass that has a suitable hashCode/equals method and creates the number-in-a-circle image whenever told to do so.
With this descriptor only one image is created for each number-background combination.

Furthermore the transparency of the overlay did not work (you can see the white corners):
grafik

With this change it is now:
grafik

File: org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java (hunk 1 above)
Comment (HannesWell) [reply org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java:48]: That's right, but the `resourceManager` is constructed using the two argument constructor:
`resourceManager = new LocalResourceManager(JFaceResources.getResources(), parentShell);`

So when the `parentShell` is disposed the `resourceManager` is disposed too.

**Comment by jukzi** [comment]
> missing/improper `hashcode/equals`.

why don't you fix that?

File: org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java (hunk 1 above)
Comment (jukzi) [reply org.eclipse.tips.ui/src/org/eclipse/tips/ui/internal/TipDialog.java:48]: neat

**Review by jukzi — APPROVED** [review]

**Pull request merged**